db/migrations/20190711091453_create_flip_table.sql

```diff
@@ -232,32 +219,24 @@ $$
 CREATE OR REPLACE FUNCTION maker.insert_updated_flip_lot() RETURNS TRIGGER AS
 $$
 BEGIN
-    WITH created AS (
-        SELECT created
-        FROM maker.flip
-        WHERE flip.bid_id = NEW.bid_id
-        ORDER BY flip.block_number
-        LIMIT 1
-    ),
-         diff_block AS (
-             select block_number, hash, block_timestamp
-             FROM public.headers
-             WHERE id = NEW.header_id
-         )
+    WITH diff_block AS (
+        SELECT block_number, block_timestamp
+        FROM public.headers
+        WHERE id = NEW.header_id
+    )
     INSERT
-    INTO maker.flip(bid_id, address_id, block_number, block_hash, lot, guy, tic, "end", bid, gal, tab, updated,
-                    created)
-    VALUES (NEW.bid_id, NEW.address_id, (SELECT block_number FROM diff_block), (SELECT hash FROM diff_block), NEW.lot,
-            (SELECT get_latest_flip_bid_guy(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_tic(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_end(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_bid(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_gal(NEW.bid_id)),
-            (SELECT get_latest_flip_bid_tab(NEW.bid_id)),
+    INTO maker.flip (bid_id, address_id, block_number, lot, guy, tic, "end", bid, gal, tab, updated, created)
+    VALUES (NEW.bid_id, NEW.address_id, (SELECT block_number FROM diff_block), NEW.lot,
+            get_latest_flip_bid_guy(NEW.bid_id),
+            get_latest_flip_bid_tic(NEW.bid_id),
+            get_latest_flip_bid_end(NEW.bid_id),
+            get_latest_flip_bid_bid(NEW.bid_id),
+            get_latest_flip_bid_gal(NEW.bid_id),
+            get_latest_flip_bid_tab(NEW.bid_id),
             (SELECT api.epoch_to_datetime(block_timestamp) FROM diff_block),
-            (SELECT created FROM created))
-    ON CONFLICT (bid_id, block_number) DO UPDATE SET lot = NEW.lot;
-    return NEW;
+            flip_bid_time_created(NEW.address_id, NEW.bid_id))
+    ON CONFLICT (block_number, bid_id, address_id) DO UPDATE SET lot = NEW.lot;
+    RETURN NEW;
```

Comment:
According to [the docs](https://www.postgresql.org/docs/11/plpgsql-trigger.html)

> A trigger function must return either NULL or a record/row value having exactly the structure of the table the trigger was fired for.

So `RETURN NULL` would work just as well here (though I don't think it would work to omit the return entirely). I don't know of any reason to pick one over the other, so I randomly settled on `NEW`. 🤷‍♂